ui/src/app/shared/components/editable-panel/editable-section.tsx

```diff
@@ -57,17 +60,32 @@ export class EditableSection<T = {}> extends React.Component<EditableSectionProp
                             right: this.props.isTopSection ? (this.props.collapsible ? '4.5em' : '3.5em') : this.props.collapsible ? '1.0em' : '0em'
                         }}>
                         {!this.state.isEditing && (
-                            <button
-                                key={'edit_button_' + this.props.uniqueId}
-                                onClick={() => {
-                                    this.setState({isEditing: true});
-                                    this.props.updateButtons(true);
-                                    this.props.onModeSwitch();
-                                }}
-                                disabled={this.props.disabledState}
-                                className='argo-button argo-button--base'>
-                                Edit
-                            </button>
+                            <div className='editable-panel__buttons-relative-button'>
+                                <button
+                                    key={'edit_button_' + this.props.uniqueId}
+                                    onClick={() => {
+                                        this.setState({isEditing: true});
+                                        this.props.updateButtons(true);
+                                        this.props.onModeSwitch();
+                                    }}
+                                    disabled={this.props.disabledState}
+                                    className='argo-button argo-button--base'>
+                                    Edit
+                                </button>{' '}
+                                {this.props.isTopSection && this.props.deleteSource && (
+                                    <button
+                                        key={'delete_button_' + this.props.uniqueId}
+                                        onClick={() => {
+                                            this.props.deleteSource();
+                                        }}
+                                        disabled={this.props.disabledDelete}
+                                        className='argo-button argo-button--base'>
+                                        {helpTip('Delete the source from the sources field')}
+                                        <span style={{marginRight: '8px'}} />
+                                        Delete
+                                    </button>
+                                )}
+                            </div>
                         )}
```

Comment:
I saw this error before when I added the initial mutli-source app support, so I added the keys. I can't seem to reproduce this problem.  Perhaps we can fix this in another issue since the issue isn't blocking this particular feature.